### PR TITLE
[Merged by Bors] - feat: `withDensity` of an s-finite measure is s-finite

### PR DESCRIPTION
**Pull request by RemyDegenne**
---

[Open in Gitpod](https://gitpod.io/from-referrer/)

### CODE REVIEW

File: Mathlib/MeasureTheory/Measure/WithDensity.lean [hunk 1]
```diff
@@ -585,6 +585,55 @@ lemma SigmaFinite.withDensity_ofReal [SigmaFinite μ] {f : α → ℝ} (hf : AEM
...
+      split_ifs with h <;> simp [h]
+    infer_instance
+  have h2 : SigmaFinite (μ.withDensity (sᶜ.indicator f)) := by
+    refine SigmaFinite.withDensity_of_ne_top' (hf.indicator hs.compl).aemeasurable (fun x ↦ ?_)
+    simp only [Set.indicator_apply, Set.mem_compl_iff, Set.mem_setOf_eq, ite_not, ne_eq, s]
+    split_ifs with h <;> simp [h]
+  infer_instance
+
+/-- If `μ` is s-finite, then `μ.withDensity f` is s-finite. -/
+lemma sFinite_withDensity (μ : Measure α) [SFinite μ] {f : α → ℝ≥0∞} (hf : AEMeasurable f μ) :
```
Comment (JasonKYi) [inline Mathlib/MeasureTheory/Measure/WithDensity.lean:622]: I think enabling dot notation with `Measure.sFinite_withDensity` would be nice

**Review by sgouezel — COMMENTED** [review]
You have an assumption that `f` is ae-measurable in your last lemma. Do you think it is really necessary? Sketch of proof without it (I haven't checked the details, so maybe I am saying nonsense): `μ.withDensity f` is absolutely continuous with respect to `μ`, which is absolutely continuous wrt to a finite measure rho. By Radon-Nikodym, one may write `μ.withDensity f` as a measure with density wrt rho, with some measurable density. And then you can apply your first lemma.

**Comment by sgouezel** [comment]
But do you agree that in the end the result should be true without any assumption? In this case, it could we worth adding a TODO in the docstring, and using a lemma name showing more clearly that it is an auxiliary statement towards a more general result. If you only use it with measurable functions, I am also not sure it's worth stating the auxiliary result for ae-measurable functions.

**Comment by RemyDegenne** [comment]
Yes I agree that it is true more generally! I will add a TODO and keep only the result for measurable instead of ae-measurable for now.
This auxiliary lemma will then be used to prove that every measure absolutely continuous with respect to an s-finite measure is s-finite, and at that point I'll come back to this to add an instance for s-finiteness of `μ.withDensity f`.

File: Mathlib/MeasureTheory/Measure/WithDensity.lean (hunk 1 above)
Comment (RemyDegenne) [reply Mathlib/MeasureTheory/Measure/WithDensity.lean]: Following the suggestion of @sgouezel I am turning that into an auxiliary lemma for a future instance, which means that it should almost never be used, so dot notation will not be useful.

**Comment by sgouezel** [comment]
bors r+
Thanks!

**Pull request closed without merging**